src/node_file.cc

```diff
@@ -2727,6 +2730,290 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
+
+    if (!file_path_url) {
+      env->isolate()->ThrowException(
+          ERR_INVALID_URL(env->isolate(), "Invalid URL"));
+
+      return;
+    }
+
+    if (!FileURLToPath(env, file_path_url.value(), initial_file_path)) return;
+
+    FromNamespacedPath(&initial_file_path);
+
+    for (int i = 0; i < BindingData::legacy_main_extensions_with_main_end;
+         i++) {
+      file_path = initial_file_path + BindingData::legacy_main_extensions[i];
+
+      switch (FilePathIsFile(env, file_path)) {
+        case BindingData::FilePathIsFileReturnType::kIsFile:
+          return args.GetReturnValue().Set(i);
+        case BindingData::FilePathIsFileReturnType::kIsNotFile:
+          continue;
+        case BindingData::FilePathIsFileReturnType::
+            kThrowInsufficientPermissions:
+          // the default behavior when do not have permission is to return
+          // and exit the execution of the method as soon as possible
+          // the internal function will throw the exception
+          return;
+        default:
+          UNREACHABLE();
+      }
+    }
+  }
+
+  file_path_url =
+      ada::parse<ada::url_aggregator>("./index", &package_json_url.value());
+
+  if (!file_path_url) {
+    env->isolate()->ThrowException(
+        ERR_INVALID_URL(env->isolate(), "Invalid URL"));
+
+    return;
+  }
+
+  if (!FileURLToPath(env, file_path_url.value(), initial_file_path)) return;
+
+  FromNamespacedPath(&initial_file_path);
+
+  for (int i = BindingData::legacy_main_extensions_with_main_end;
+       i < BindingData::legacy_main_extensions_package_fallback_end;
+       i++) {
+    file_path = initial_file_path + BindingData::legacy_main_extensions[i];
+
+    switch (FilePathIsFile(env, file_path)) {
+      case BindingData::FilePathIsFileReturnType::kIsFile:
+        return args.GetReturnValue().Set(i);
+      case BindingData::FilePathIsFileReturnType::kIsNotFile:
+        continue;
+      case BindingData::FilePathIsFileReturnType::kThrowInsufficientPermissions:
+        // the default behavior when do not have permission is to return
+        // and exit the execution of the method as soon as possible
+        // the internal function will throw the exception
+        return;
+      default:
+        UNREACHABLE();
+    }
+  }
+
+  std::string module_path;
+  std::string module_base;
+
+  if (!FileURLToPath(env, package_json_url.value(), module_path)) return;
+
+  if (args.Length() >= 3 && args[2]->IsString()) {
+    Utf8Value utf8_base_path(env->isolate(), args[2].As<String>());
+    auto base_url =
+        ada::parse<ada::url_aggregator>(utf8_base_path.ToStringView());
+
+    if (!base_url) {
+      env->isolate()->ThrowException(
+          ERR_INVALID_URL(env->isolate(), "Invalid URL"));
+
+      return;
+    }
+
+    if (!FileURLToPath(env, base_url.value(), module_base)) return;
+  } else {
+    std::string received = args.Length() >= 3
+                               ? "How to Serialize v8::Value to a type?"
```

Comment:
@anonrig Do you know how I can get the type of v8::Value easily here?

I need to throw this error to keep the behavior consistent, this error is from `fileURLToPath` and it can happen when `base` is undefined or other value.

Reply:
I talked with Rafael, his suggestion was just to remove the "Receive " part since we cannot get easily the V8::Value type.